### PR TITLE
fix: correct handling of placeholder types

### PR DESCRIPTION
**Pull request by kevin-greene-ck**
* There was an issue where the expected type of a placeholder was not
  being passed through the function that read values. This cause values
  to be potentially parsed incorrectly.

### CODE REVIEW

File: src/main/utils/config.ts [hunk 1]
```diff
@@ -95,14 +95,20 @@ export async function readValueForType(raw: string, type: ObjectType): Promise<a
...
 
                 case 'boolean':
-                    return Promise.resolve(raw === 'true')
+                    if (raw === 'true') {
+                        return Promise.resolve(true)
+                    } else if (raw === 'false') {
+                        return Promise.resolve(false)
+                    } else {
+                        throw new DynamicConfigInvalidType(key, type)
+                    }
```
Comment (mkantor) [inline src/main/utils/config.ts:104]: Nice, I like how you tightened up the string => boolean parsing. The old behavior was pretty sketchy.

File: src/main/DynamicConfig.ts [hunk 2]
```diff
@@ -345,7 +345,7 @@ export class DynamicConfig implements IDynamicConfig {
             return this.buildDefaultForPlaceholder(placeholder)
 
         } else {
-            return resolver.get(placeholder.key).then(
+            return resolver.get(placeholder.key, placeholder.type).then(
```
Comment (mkantor) [inline src/main/DynamicConfig.ts:349]: Does the `resolver.get()` call in `getValueFromResolver` need similar treatment?  Currently it doesn't have any handling for config value types.

**Review by mkantor — COMMENTED** [review]
Thanks for the super quick turnaround! I haven't tested this out yet, but it passes a basic eyeball-check.

I left one question but it might just be a misunderstanding of the code on my part.

File: src/main/DynamicConfig.ts (hunk 2 above)
Comment (kevin-greene-ck) [reply src/main/DynamicConfig.ts:349]: Good catch. That method is really for getting raw values from the remotes so it doesn't have access to the placeholder data. I added an optional param to pass the expected type through.

**Review by mkantor — APPROVED** [review]

**Pull request merged**